### PR TITLE
Plugin tweaks

### PR DESCRIPTION
**Pull request by nwmac**
Various minor improvements to UI Plugins.

- Renamed ot 'Extensions'
- Fix LazyImage so that the icon updates when src changes (otherwise icon in preview panel does not update correctly)
- Fix for plugin does not show up if CR exists but load failed
- Add support for systemDefaultRegistry
- Improvements to the info panel 
- Fixed issue with developer load of plugins with hyphen character in the name
- Improve detection of extension support by not using the schema, but looking for the service
- Update the placeholder icon
- Fixed issue where plugins would not load after fresh local login or local login after logout
- Added checks in upfront load to check for plugin compatibility

### CODE REVIEW

File: shell/plugins/plugin.js [hunk 1]
```diff
@@ -28,11 +28,12 @@ export default async(context) => {
   }
 
   if (loadPlugins) {
-    const { store, $plugin } = context;
+    // TODO: Get rancher version using the new API (can't use setting as we have not loading the store)
```
Comment (richard-cox) [inline shell/plugins/plugin.js:31]: if we use some kind of $axios request here it can also be used for `${ UI_PLUGIN_BASE_URL }/index.json`

**Review by richard-cox — CHANGES_REQUESTED** [review]
- Plugin page action button doesn't look great in dark mode 
  - image
- Plugin slide out chart readme has white square bottom right when in dark mode
  - image
- After enabling the extention operator i switched to preferences page and returned to the extention page and get a forever spinning loading indicator and the error below. after refresh is fine. guessing the install may still be in progress?
  -
   ```
   console.js:31 Error in fetch(): Error: Unknown schema for type: catalog.cattle.io.uiplugin
    at Object.<anonymous> (getters.js:243)
    ```
- Enhancement - Do we need docs or onscreen advice to add the repos in the local cluster?
- Enhancement - Show the repo where the plugin comes from in the plugin card
- Enhancement - It's not clear when clicking `Install` on a card that you can select a previous one. For example the Homepage card states 0.2.0 and insinuates clicking on install will install 0.2.0
- Enhancement - Make the name of the plugin more distinct from the title text (do we do this for other modals? 
  - image
- Enhancement - In the apps --> charts list we show icons with a white background to cater for dark icons on dark backgrounds in dark mode. could we do something similar for the extension cards? 
  - image
- After installing 1.0 of the homepage i tried to update to 1.0 (which might be a valid case? not sure). The homepage card showed a forever spinning `Installing` notification. I then refreshed and tried updating to 2.0 and got the same result
  - The local cluster's operations show both as state green 'Active'  (as per install of extension operator)
  - It looks like the action completes ok, it's just the state in the card that doesn't get updated
  - Enhancement - We should provide a way for the user to understand progress and failure reason of installation 
- Enhancement - Add console logging for removePlugin and loadAsync actions (i have a feeling we'll need to know what plugins are misbehaving)
- After setting up github auth provider and logging out, trying to log in via github throws the user back to the login page. i think this due to a 401 for `api/v1/namespaces/cattle-ui-plugin-system/services/http:ui-plugin-operator:80/proxy/index.json`


I haven't tested some things, will try on Monday 
- will hack in some versions to test compatibility 
- i built and 'dev' loaded the harvester plugin (which shows up as third-party). i'd like to test with the normal harvester approach and validate that still works (and also update the package json to ensure it's not marked as third party)

File: shell/plugins/plugin.js (hunk 1 above)
Comment (nwmac) [reply shell/plugins/plugin.js:31]: The index.json request need to be authenticated - not sure a raw axios request will work?

File: shell/assets/translations/en-us.yaml [hunk 2]
```diff
@@ -3674,78 +3674,84 @@ plugins:
...
     versions: Versions
     versionError: Could not load version information
   empty:
-    all: No UI Plugins either installed nor available
-    available: No UI Plugins available
-    installed: No UI Plugins installed
-    updates: No UI Plugin updates available
-  loadError: An error occurred loading the code for this plugin
-  helmError: "An error occurred installing the plugin via Helm"
+    all: No Extensions are either installed nor available
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/assets/translations/en-us.yaml:3703]: Done

**Review by richard-cox — APPROVED** [review]

**Pull request merged**